graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -629,6 +654,7 @@ def _compile_root_ast_to_ir(schema, ast, type_equivalence_hints=None):
         'inputs': dict(),
         'location_types': dict(),
         'type_equivalence_hints': type_equivalence_hints or dict(),
+        'marked_location_stack': []
```

Comment:
I know the existing code doesn't do this (and that's a problem of its own), but let's start adding a short explanation of what each thing in the `context` does, what it is intended for, and what problem it solves. You can add this as a multi-line comment before the key.